br/pkg/lightning/backend/external/util.go

```diff
@@ -64,9 +63,7 @@ func seekPropsOffsets(
 	for i := range offsetsPerFile {
 		offsetsPerFile[i] = make([]uint64, len(starts))
 	}
-	// Record first key if it is smaller than first key of "starts" key argument for
-	// each file, and check all files afterward.
-	firstKeyTooSmallCheckers := make([]kv.Key, len(paths))
+
```

Comment:
also need to fix the unit tests and change the comments of seekPropsOffsets

Reply:
```diff
diff --git a/br/pkg/lightning/backend/external/util.go b/br/pkg/lightning/backend/external/util.go
index d880ca04e9..2ee8d9942b 100644
--- a/br/pkg/lightning/backend/external/util.go
+++ b/br/pkg/lightning/backend/external/util.go
@@ -40,11 +40,6 @@ import (
 // file reader, read, parse and skip few smaller keys, and then locate the needed
 // data.
 //
-// To avoid potential data loss, it also checks at least one statistic file has a
-// key larger than or equal to the start key. If not, we are afraid that some
-// paths are missing, and the data between [start key, min(first key of
-// statistic files)) are lost.
-//
 // Caller can specify multiple ascending keys and seekPropsOffsets will return
 // the offsets list per file for each key.
 func seekPropsOffsets(
diff --git a/br/pkg/lightning/backend/external/util_test.go b/br/pkg/lightning/backend/external/util_test.go
index 1800a27d23..163d908a2e 100644
--- a/br/pkg/lightning/backend/external/util_test.go
+++ b/br/pkg/lightning/backend/external/util_test.go
@@ -87,15 +87,17 @@ func TestSeekPropsOffsets(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, [][]uint64{{10, 20}, {30, 20}}, got)
 
-	_, err = seekPropsOffsets(ctx, []kv.Key{[]byte("key0")}, []string{file1, file2}, store)
-	require.ErrorContains(t, err, "start key 6b657930 is too small for stat files [/test1 /test2]")
+	got, err = seekPropsOffsets(ctx, []kv.Key{[]byte("key0")}, []string{file1, file2}, store)
+	require.NoError(t, err)
+	require.Equal(t, [][]uint64{{0, 0}}, got)
 
 	got, err = seekPropsOffsets(ctx, []kv.Key{[]byte("key1")}, []string{file1, file2}, store)
 	require.NoError(t, err)
 	require.Equal(t, [][]uint64{{10, 0}}, got)
 
-	_, err = seekPropsOffsets(ctx, []kv.Key{[]byte("key0"), []byte("key1")}, []string{file1, file2}, store)
-	require.ErrorContains(t, err, "start key 6b657930 is too small for stat files [/test1 /test2]")
+	got, err = seekPropsOffsets(ctx, []kv.Key{[]byte("key0"), []byte("key1")}, []string{file1, file2}, store)
+	require.NoError(t, err)
+	require.Equal(t, [][]uint64{{0, 0}, {10, 0}}, got)
 
 	got, err = seekPropsOffsets(ctx, []kv.Key{[]byte("key999")}, []string{file1, file2}, store)
 	require.NoError(t, err)
```